src/snowflake/snowpark/udf.py

```diff
@@ -897,3 +915,115 @@ def _do_register_udf(
                 )
 
         return UserDefinedFunction(func, return_type, input_types, udf_name)
+
+    def __do_register_udf(
+        self,
+        callableProperties: CallableProperties,
+        api_call_source: str,
+        stage_location: Optional[str] = None,
+        statement_params: Optional[Dict[str, str]] = None,
+        from_pandas_udf_function: bool = False,
+        skip_upload_on_content_match: bool = False,
+    ) -> UserDefinedFunction:
+
+        (
+            udf_name,
+            is_pandas_udf,
+            is_dataframe_input,
+            return_type,
+            input_types,
+        ) = process_registration_inputs(
+            session=self._session,
+            object_type=callableProperties.object_type,
+            func=callableProperties.func,
+            return_type=callableProperties.raw_return_type,
+            input_types=callableProperties.raw_input_types,
+            name=callableProperties.raw_name,
+        )
+
+        callableProperties.setValidatedObjectName(udf_name)
+        callableProperties.setValidatedReturnType(return_type)
+        callableProperties.setValidatedInputTypes(input_types)
```

Comment:
Really not loving the fact that the input param sneakily gets mutated here. Should this logic be moved to methods on the CallableProperties object itself? It would make the mutation a lot more obvious IMHO. Something like:
`callableProperties.processRegistrationInputs(session)` instead of the logic above (and similarly for the other pieces)

Reply:
I understand that, but from a maintenance POV, modifying an object that's passed as a parameter in the middle of other logic is usually a surprising side effect. The way this method is named, it says "perform the registration using the provided properties", but then the object gets mutated along the way.